src/main/java/it/gov/pagopa/pu/classification/service/BalanceService.java

```diff
@@ -1,27 +1,100 @@
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
 import org.springframework.stereotype.Service;
 
+import java.time.LocalDate;
+import java.util.Set;
+
 @Service
 @Slf4j
 public class BalanceService {
 
-  private final BalanceUnmashallerService balanceUnmashallerService;
+  private static final String OPERATING_YEAR = String.valueOf(LocalDate.now().getYear());
+  private static final AssessmentsRegistryStatus ASSESSMENTS_REGISTRY_STATUS = AssessmentsRegistryStatus.ACTIVE;
+  private static final String BALANCE_AMOUNT_DEFAULT_VALUE = "TOTALE";
+
+  private final BalanceUnmarshallerService balanceUnmashallerService;
+  private final BalanceDefaultMarshallingService balanceDefaultMarshallingService;
+  private final AssessmentsRegistryRepository assessmentsRegistryRepository;
 
-  public BalanceService(BalanceUnmashallerService balanceUnmashallerService) {
+  public BalanceService(BalanceUnmarshallerService balanceUnmashallerService, BalanceDefaultMarshallingService balanceDefaultMarshallingService, AssessmentsRegistryRepository assessmentsRegistryRepository) {
     this.balanceUnmashallerService = balanceUnmashallerService;
+    this.balanceDefaultMarshallingService = balanceDefaultMarshallingService;
+    this.assessmentsRegistryRepository = assessmentsRegistryRepository;
   }
 
-  public Boolean isBalanceValid(String balance){
+  public Boolean isBalanceValid(String balance) {
     try {
-      balanceUnmashallerService.unmarshal(balance);
+      validate(balance);
       log.info("The balance value is formally valid");
       return Boolean.TRUE;
-    } catch (InvalidValueException invalidValueException){
+    } catch (InvalidValueException invalidValueException) {
       log.info("The balance value is not valid: {}", invalidValueException.getMessage());
       return Boolean.FALSE;
     }
   }
+
+  private void validate(String balance) {
+    try {
+      log.info("Validating balance value with default structure");
+      balanceDefaultMarshallingService.unmarshal(balance);
+    } catch (InvalidValueException invalidValueException) {
+      log.info("Validating balance value with actual structure");
+      balanceUnmashallerService.unmarshal(balance);
+    }
+  }
+
+  public String getBalanceByAssessmentRegistry(Long organizationId, String debtPositionTypeOrgCode) {
+    log.info("Retrieving balance from AssessmentsRegistry with orgId[{}], debtPositionTypeCode[{}], operatingYear [{}], status [{}]",
+      organizationId, debtPositionTypeOrgCode, OPERATING_YEAR, ASSESSMENTS_REGISTRY_STATUS);
+    Page<AssessmentsRegistry> assessmentsRegistries = assessmentsRegistryRepository.findAssessmentsRegistriesByFilters(
+      organizationId,
+      Set.of(debtPositionTypeOrgCode),
+      null, null, null, null, null, null,
+      OPERATING_YEAR,
+      ASSESSMENTS_REGISTRY_STATUS,
+      PageRequest.of(0, 5));
+
+    int size = assessmentsRegistries.getSize();
+    if (size > 1) {
```

Comment:
probably we could accept many AssessmentsRegistries... we will have many Balance for each officeCode retrieved..

CC: @robertopasetti , @RiccardoGiuliani

Reply:
At most, we will have just one record ACTIVE